mungegithub/mungers/flake-manager.go

```diff
@@ -116,17 +116,80 @@ func (p *FlakeManager) AddFlags(cmd *cobra.Command, config *github.Config) {}
 // Munge is unused by this munger.
 func (p *FlakeManager) Munge(obj *github.MungeObject) {}
 
+// Look through all issues filed about this test.
+// If foundIn is > 0, then the particular flake was found in that issue.
+// All open issues for this test are returned in updatableIssues.
+func (p *FlakeManager) findPreviousIssues(f cache.Flake) (foundIn int, updatableIssues []*github.MungeObject, err error) {
+	possibleIssues := p.finder.AllIssuesForKey(string(f.Test))
+	for _, previousIssue := range possibleIssues {
+		obj, err := p.config.GetObject(previousIssue)
+		if err != nil {
+			return 0, nil, fmt.Errorf("error getting object for %v: %v", previousIssue, err)
+		}
+		isRecorded, err := p.isRecorded(obj, f)
+		if err != nil {
+			return 0, nil, fmt.Errorf("error checking whether flake %v is recorded in issue %v: %v", f, previousIssue, err)
+		}
+		if isRecorded {
+			p.alreadySyncedFlakes[f] = previousIssue
+			foundIn = previousIssue
+			// keep going since we may want to close dups
+		}
+		if obj.Issue.State != nil && *obj.Issue.State == "open" {
+			updatableIssues = append(updatableIssues, obj)
+		}
+	}
+	return foundIn, updatableIssues, nil
+}
+
+// Close all of the dups.
+func (p *FlakeManager) markAsDups(dups []*github.MungeObject, of int) error {
+	// Somehow we got duplicate issues all open at once.
+	// Close all of the older ones.
+	for _, dup := range dups {
+		if err := dup.CloseIssuef("This is a duplicate of #%v; closing", of); err != nil {
+			return fmt.Errorf("failed to close %v as a dup of %v: %v", *dup.Issue.Number, of, err)
+		}
+	}
+	return nil
+}
+
 func (p *FlakeManager) syncFlake(f cache.Flake) error {
 	if _, ok := p.alreadySyncedFlakes[f]; ok {
 		return nil
 	}
-	if n, ok := p.finder.IssueForKey(string(f.Test)); ok {
-		if err := p.updateIssue(n, f); err != nil {
+
+	foundIn, updatableIssues, err := p.findPreviousIssues(f)
+	if err != nil {
+		return err
+	}
+
+	// Close dups if there are multiple open issues
```

Comment:
Yes. This shouldn't really happen unless the merge queue misbehaves.

On Thu, May 26, 2016 at 11:33 AM, Eric Paris <notifications@github.com>
wrote:

> In mungegithub/mungers/flake-manager.go
> <https://github.com/kubernetes/contrib/pull/1061#discussion_r64797021>:
>
> >  func (p *FlakeManager) syncFlake(f cache.Flake) error {
> >  	if _, ok := p.alreadySyncedFlakes[f]; ok {
> >  		return nil
> >  	}
> > -	if n, ok := p.finder.IssueForKey(string(f.Test)); ok {
> > -		if err := p.updateIssue(n, f); err != nil {
> > +
> > +	foundIn, updatableIssues, err := p.findPreviousIssues(f)
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	// Close dups if there are multiple open issues
>
> am I understanding right? we pick the newest/highest number issue and mark
> the rest as a dup of the newest one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1061/files/e481c7373b54ad4eee2214188a0f5adfa809b3fb#r64797021>
>